src/utilities/did/did.ts

```diff
@@ -1,54 +1,80 @@
-import {
-  ConfigService,
+import type {
   Did,
   DidDocument,
-  DidEncryptionKey,
-  DidUri,
-} from '@kiltprotocol/sdk-js';
+  KiltKeyringPair,
+  UriFragment,
+  VerificationMethod,
+} from '@kiltprotocol/types';
+
+import { DidResolver } from '@kiltprotocol/sdk-js';
+import {
+  didKeyToVerificationMethod,
+  getFullDid,
+  parse,
+  parseDocumentFromLightDid,
+} from '@kiltprotocol/did';
+
+import { blake2AsHex } from '@polkadot/util-crypto';
 
 import { useAsyncValue } from '../useAsyncValue/useAsyncValue';
 
-export function isFullDid(did: DidUri | undefined): boolean {
+export function isFullDid(did: Did | undefined): boolean {
   if (!did) {
     // maybe the DID was deleted from the blockchain
     return false;
   }
-  return Did.parse(did).type === 'full';
+  return parse(did).type === 'full';
 }
 
-export async function getFullDidDocument(did: DidUri): Promise<DidDocument> {
-  const api = ConfigService.get('api');
-  return Did.linkedInfoFromChain(await api.call.did.query(Did.toChain(did)))
-    .document;
+export async function getFullDidDocument(did: Did): Promise<DidDocument> {
+  const { didDocument } = await DidResolver.resolve(did, {});
+  if (!didDocument) {
+    throw new Error('Unable to resolve DID');
```

Comment:
There's even resolution metadata which gives you the exact reason why the DID could not be resolved. Maybe it would be useful to add it to the error thrown.